Userland/Services/SystemServer/Service.cpp

```diff
@@ -380,6 +380,9 @@ ErrorOr<void> Service::determine_account(int fd)
 
 Service::~Service()
 {
+    if (m_pid == -1)
```

Comment:
For a multi-instance services `m_pid == -1` but you will still want to remove the `notifier_socker`, right?
Sorry, I don't have time to look more into this right now.